test/formulas/index.js

```diff
@@ -1,11 +1,57 @@
-require('../operators/operators');
-require('./math/math');
-require('./trigonometry/trigonometry');
-require('./text/text');
-require('./reference/reference');
-require('./information/information');
-require('./statistical/statistical');
-require('./date/date');
-require('./engineering/engineering');
-require('./logical/logical');
-require('./web/web');
+const FormulaError = require('../../formulas/error');
+const {FormulaParser} = require('../../grammar/hooks');
+const json = require('./TESTS.json')
+const expect = require('chai').expect;
+const sinon = require('sinon')
+
+const test = function(testName){
+    const date = new Date(2000, 1, 1)
+    let data = json[testName]["sheet"],
+        formula = json[testName]["formula"],
+        expected = json[testName]["expectedValue"]
+        clock =  sinon.useFakeTimers(date.getTime());
+    
+    if(data){
+        data.forEach((row, rowIndex) => row.forEach((value, colIndex) => {
+            if(value === "undefined")
+                data[rowIndex][colIndex] = undefined
+            if(FormulaError.stringToError(value))
+                data[rowIndex][colIndex] = FormulaError.stringToError(value);
```

Comment:
Looking at how `stringToError()` is used, you might be able to remove it entirely and instead just check `FormulaError.errorMap.has(value)` and/or `FormulaError.errorMap.get(value)`

`FormulaError.errorMap` stores all the "generic" errors that are checked in `stringToError()`.